src/language-js/parser-babylon.js

```diff
@@ -4,71 +4,99 @@ const createError = require("../common/parser-create-error");
 const hasPragma = require("./pragma").hasPragma;
 const locFns = require("./loc");
 
+function babylonOptions(extraOptions, extraPlugins) {
+  return Object.assign(
+    {
+      sourceType: "module",
+      allowAwaitOutsideFunction: true,
+      allowImportExportEverywhere: true,
+      allowReturnOutsideFunction: true,
+      allowSuperOutsideMethod: true,
+      plugins: [
+        "jsx",
+        "flow",
+        "doExpressions",
+        "objectRestSpread",
+        "classProperties",
+        "exportDefaultFrom",
+        "exportNamespaceFrom",
+        "asyncGenerators",
+        "functionBind",
+        "functionSent",
+        "dynamicImport",
+        "numericSeparator",
+        "importMeta",
+        "optionalCatchBinding",
+        "optionalChaining",
+        "classPrivateProperties",
+        ["pipelineOperator", { proposal: "minimal" }],
+        "nullishCoalescingOperator",
+        "bigInt",
+        "throwExpressions"
+      ].concat(extraPlugins)
+    },
+    extraOptions
+  );
+}
+
 function parse(text, parsers, opts) {
   // Inline the require to avoid loading all the JS if we don't use it
   const babylon = require("@babel/parser");
 
-  const babylonOptions = {
-    sourceType: "module",
-    allowAwaitOutsideFunction: true,
-    allowImportExportEverywhere: true,
-    allowReturnOutsideFunction: true,
-    allowSuperOutsideMethod: true,
-    plugins: [
-      "jsx",
-      "flow",
-      "doExpressions",
-      "objectRestSpread",
-      "decorators-legacy",
-      "classProperties",
-      "exportDefaultFrom",
-      "exportNamespaceFrom",
-      "asyncGenerators",
-      "functionBind",
-      "functionSent",
-      "dynamicImport",
-      "numericSeparator",
-      "importMeta",
-      "optionalCatchBinding",
-      "optionalChaining",
-      "classPrivateProperties",
-      ["pipelineOperator", { proposal: "minimal" }],
-      "nullishCoalescingOperator",
-      "bigInt",
-      "throwExpressions"
-    ]
-  };
+  const combinations = [
```

Comment:
We probably need a `guess` function (similar to `isProbablyJsx` in `parser-typescript.js`) to try not to do unnecessary parsing.

Reply:
I thought about it, trying to find a `@deco` before/after an export